rockcraft.yaml

```diff
@@ -0,0 +1,203 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: discourse
+summary: Discourse rock
+description: Discourse OCI image for the Discourse charm
+base: ubuntu:20.04
+license: Apache-2.0
+version: "1.0"
+platforms:
+  amd64:
+parts:
+  debug-tools:
+    plugin: nil
+    stage-packages:
+      - git
+      - coreutils
+      - bash
+      - curl
+      - vim
+  get-packages:
+    plugin: nil
+    overlay-packages:
+      - brotli
+      - ca-certificates
+      - curl
+      - gettext-base
+      - gifsicle
+      - git
+      - imagemagick
+      - jhead
+      - jpegoptim
+      - libjpeg-turbo-progs
+      - libpq-dev
+      - libssl-dev
+      - libxml2-dev
+      - libxslt1-dev
+      - libz-dev
+      - optipng
+      - pngquant
+      - postgresql-client
+      - postgresql-client-common
+      - redis-tools
+      - ruby2.7
+      - ruby2.7-dev
+      - tzdata
+      - ubuntu-dev-tools
+      - zlib1g-dev
+  get-discourse:
+    plugin: dump
+    source: https://github.com/discourse/discourse.git
+    source-depth: 1
+    source-tag: v2.8.14
+    source-type: git
+    override-build: |
+      craftctl default
+      mkdir -p tmp/backups/default
+      mkdir -p public/backups/default
+      mkdir -p public/uploads/default
+      mkdir -p log/production.log
+      mkdir -p srv/discourse/app/bin
+      touch log/production.log
+      touch log/unicorn-stderr.log
+    organize:
+      "*.*": srv/discourse/app/
+      "*": srv/discourse/app/
+      ".*": srv/discourse/app/
+  get-solved-plugin-source:
+    plugin: dump
+    source: https://github.com/discourse/discourse-solved.git
+    source-depth: 1
+    source-commit: d6c8089ca38611b09a8edb29d64f359bcef11f11
+    after:
+      - get-discourse
+    organize:
+      "*": srv/discourse/app/plugins/discourse-solved/
```

Comment:
Perhaps for when the rockfile is more stable, it would be good to include some documentation with reasoning for the different parts here, e.g., why installing from git etc

Reply:
I added some comments.  
As for why installing from git: because the dockerfile used to do it that way (I didn't want to change things if not needed) and I'm not even sure there's another way to get discourse ?